repose-aggregator/core/repose-core/src/test/scala/org/openrepose/powerfilter/ReposeRoutingServletTest.scala

```diff
@@ -0,0 +1,230 @@
+      (null,  SC_OK,                       new IOException()),
+      (null,  SC_OK,                       new IOException().initCause(new ReadLimitReachedException("too much"))),
+      ("one", SC_OK,                       null),
+      ("one", SC_SERVICE_UNAVAILABLE,      new IOException()),
+      ("one", SC_REQUEST_ENTITY_TOO_LARGE, new IOException().initCause(new ReadLimitReachedException("too much"))),
+      ("two", SC_OK,                       null),
+      ("two", SC_SERVICE_UNAVAILABLE,      new IOException()),
+      ("two", SC_REQUEST_ENTITY_TOO_LARGE, new IOException().initCause(new ReadLimitReachedException("too much")))
+      // @formatter:on
+    ).foreach { case (defaultDestinationId, responseCode, dispatchError) =>
+      Seq(null, 0, 8080, 8088).foreach { port =>
+        it(s"port number: $port, route default destination: $defaultDestinationId, response code: $responseCode, dispatch error: ${Option(dispatchError).flatMap(e => Option(e.getCause).map(ie => ie.getMessage).orElse(Option("None"))).getOrElse("Null")}") {
+          val request = new MockHttpServletRequest("POST", "/this/is/the/requestURI")
+          request.setScheme("http")
+          request.setServerName(DefaultNodeName)
+          request.setServerPort(8080)
+          val response = new MockHttpServletResponse()
+          val systemModel = minimalConfiguration()
+          val destinationEndpointList = systemModel.getReposeCluster.headOption.map(_.getDestinations).map(_.getEndpoint)
+          destinationEndpointList.foreach(_.clear())
+          val destinationEndpointOne = new DestinationEndpoint()
+          destinationEndpointOne.setDefault("one".equals(defaultDestinationId))
+          destinationEndpointOne.setId("one")
+          destinationEndpointOne.setProtocol("http")
+          Option(port).foreach(p => destinationEndpointOne.setPort(p.asInstanceOf[Int]))
+          destinationEndpointList.foreach(_.add(destinationEndpointOne))
+          val destinationEndpointTwo = new DestinationEndpoint()
+          destinationEndpointTwo.setDefault("two".equals(defaultDestinationId))
+          destinationEndpointTwo.setId("two")
+          destinationEndpointTwo.setProtocol("http")
+          destinationEndpointList.foreach(_.add(destinationEndpointTwo))
+          val servletContext = mock[ServletContext]
+          val targetServletContext = mock[ServletContext]
+          val requestDispatcher = mock[RequestDispatcher]
+
+          when(servletContext.getContext(anyString())).thenReturn(targetServletContext)
+          when(targetServletContext.getRequestDispatcher(anyString())).thenReturn(requestDispatcher)
+          when(targetServletContext.getContextPath).thenReturn("")
+          mockServletConfig = new MockServletConfig(servletContext)
+
+          if (dispatchError != null) {
+            when(requestDispatcher.forward(any(classOf[ServletRequest]), any(classOf[ServletResponse]))).thenThrow(dispatchError)
+          }
```

Comment:
Oh man do I dislike conditional behavior in tests. It makes the intent of the test more difficult to grasp and expands the scope of the test. It also makes the test more difficult to maintain moving forward.

This may be a place where I find the code a little too DRY, and would prefer it be more DAMP:
https://stackoverflow.com/a/11837973

Reply:
It is easy enough to add two more boolean values to the test data, but it just seemed redundant to put in the manually calculated value.

Reply:
It would end up with:
```
false | false
false | false
false | false
true  | false
true  | false
true  | false
false | true
false | true
false | true
```

Reply:
For the dispatch error tests that are `null`, then they can be broken out to a separate test. But again, that seemed like overkill.